server/model/block.go

```diff
@@ -161,6 +168,60 @@ func (b *Block) IsValid() error {
 	return nil
 }
 
+var safeInputPattern = regexp.MustCompile(`^[a-zA-Z0-9 _-]+$`)
+
+func ValidateFileId(id string) error {
+	if id == "" {
+		return errEmptyId
+	}
+
+	if len(id) < MinIdLength {
+		return errInvalidId
+	}
+
+	if !mmModel.IsValidId(id[1:28]) {
```

Comment:
Is that :28 or :27?

Reply:
This should be 28 as we want the character from 27 characters, starting from 1 to 28.

Reply:
Its always confusing whether the last index is "including", or "up to before" 🤦

Reply:
The slice expression id[1:28] means:

- Start from index 1 (inclusive).
- Go up to index 28 (exclusive).
- This results in selecting 27 characters, from index 1 to 27.